### PR TITLE
Feature/bot registration

### PR DESCRIPTION
**Pull request by unlimit**

### CODE REVIEW

File: spec/controllers/products_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :controller do
+
+  describe "GET #create" do
+    it "returns http success" do
+      get :create
```
Comment (unlimit) [inline spec/controllers/products_controller_spec.rb]: maybe post create?

File: app/javascript/reducers/products.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { ProductsActions } from '../actions/productsActions.js'
+import update from 'immutability-helper';
+import {normalizeProducts} from '../helpers/JsonNormalizer';
+
+let products_list = default_props && default_props.developer ? normalizeProducts(default_props.developer.products) : {byAddress: {}, allIds: []}
+const initialState = {
+  inProgress: false,
+  registration_steps: {
+    load_to_ipfs: {status: 'waiting', text: 'Loading metadata to IPFS'},
```
Comment (unlimit) [inline app/javascript/reducers/products.js]: Is reducer good place to store structure like load_to_ipfs ?

File: app/javascript/actions/productsActions.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ * @param values - AI product metadata as JS object
+**/
+const addMetadata2IPFS = (values) => (dispatch) => {
+  return new Promise((resolve, reject) => {
+    dispatch(setProgressStatus('load_to_ipfs', 'running'));
+    const config = { headers: { 'content-type': 'multipart/form-data' } };
+    const formData = new FormData()
+    formData.append('file', JSON.stringify(values))
+
+    axios.post('https://ipfs.infura.io:5001/api/v0/add?pin=true', formData, config)
```
Comment (unlimit) [inline app/javascript/actions/productsActions.js]: should we move 'https://ipfs.infura.io:5001/' to configs?

File: app/javascript/blockchain/KeyTools.js [hunk 4]
```diff
@@ -18,6 +18,16 @@ class KeyTools {
     return this.web3.eth.accounts.wallet[0].address;
   }
 
+  /** Generates random valid address */
+  generateRandomAddress(){
```
Comment (unlimit) [inline app/javascript/blockchain/KeyTools.js]: Should generated address be tied to user's mnemonic? 
In this case user could have a private key for new address

File: app/models/product.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+class Product < ApplicationRecord
+  belongs_to :user
+
+  validates :name, :eth_address, :create_bot_product_tx, presence: true
+
+  def as_json(options={})
```
Comment (unlimit) [inline app/models/product.rb]: where do we use this?

File: app/javascript/actions/productsActions.js (hunk 3 above)
Comment (server-izetov) [reply app/javascript/actions/productsActions.js]: I thought about this, but if we changed IPFS services, I think we should also change source code for process data from it. That's why I leave URL in code.

File: app/javascript/blockchain/KeyTools.js (hunk 4 above)
Comment (server-izetov) [reply app/javascript/blockchain/KeyTools.js]: I think we can remove it now.

File: app/javascript/reducers/products.js (hunk 2 above)
Comment (server-izetov) [reply app/javascript/reducers/products.js]: I used this for visualisation bot registration process in UI.
Do you propose store in reducer something like this:
```js
registration_steps: {
  load_to_ipfs: 'waiting',
  approve: 'waiting',
  add_bot: 'waiting'
}
```
and `text` and `stepsOrder` move to component where we used these steps?

File: app/models/product.rb (hunk 5 above)
Comment (server-izetov) [reply app/models/product.rb]: we used this [here](https://github.com/TallaBotChain/botchain-example-service/pull/42/files#diff-4676c008b11a5480d73d4a6de01e45b9R14) and then [here](https://github.com/TallaBotChain/botchain-example-service/pull/42/files#diff-9599427925097c3c66f26ac1e0de5cadR28) And then used this as initial state for developer and products

**Pull request merged**